### PR TITLE
Resize sidebar

### PR DESCRIPTION
**Pull request by ironyman**
Addresses https://github.com/mozilla/pdf.js/issues/2072.

### CODE REVIEW

**Comment by timvandermeij** [comment]
I like the direction of this patch and it already works quite well. Some comments from playing with this a bit:

- Let's restrict the sidebar width to 50% of the total screen width. Now it's possible to even resize until beyond 100%, which should not happen. After more than 50%, the PDF file itself becomes too small, so I think 50% would be a good maximum value.

- I think the timeout until the view re-renders after resizing is a bit too large, which makes the resizing feel slow. If you resize the sidebar quickly, the cursor can end up behind the page itself until it's re-rendered.

**Comment by ironyman** [comment]
I addressed both of those issues.
But I noticed another problem. 
2017-09-26-222312_1037x146_scrot
toolbarViewerMiddle gets pushed off the toolbar when there's not enough space. What should we do about this?

**Comment by Snuffleupagus** [comment]
In addition to https://github.com/mozilla/pdf.js/pull/8957#issuecomment-332336558, there's also a couple of other issues:
 - Most importantly, since it breaks things quite badly, is that if you resize the sidebar then the viewer will look *completely* broken after the sidebar has been closed.

 - I'm finding the code somewhat difficult to follow, and my worry is that this would be very hard to actually maintain in the future. Considering that the added JS code is making *a lot* of assumptions about the structure of the HTML/CSS, making any changes down the line there would thus be difficult.

(Presumably the reason why this functionality hasn't been implemented yet, is that the way e.g. the sidebar CSS is currently written makes this very difficult to fix in a good way.)

**Comment by ironyman** [comment]
Are we OK with rewriting the sidebar CSS using flexbox? That should eliminate the need for hard coded width on every sub element of sidebarContainer.

File: web/pdf_sidebar.js [hunk 1]
```diff
@@ -454,6 +460,81 @@ class PDFSidebar {
...
+
+  resize(width) {
+    if (width < 200 || width > this.mainContainer.offsetWidth) {
+      return;
+    }
+
+    // This is the media query in viewer.css, which
+    // sets #mainContainer left to 0 on small screens.
+    // We don't want to override that.
+    var mql = window.matchMedia('all and (max-width: 840px');
```
Comment (yurydelendik) [inline web/pdf_sidebar.js]: Per http://caniuse.com/#search=matchMedia most of browsers we are supporting are not mentioned.

**Comment by yurydelendik** [comment]
> Are we OK with rewriting the sidebar CSS using flexbox?

Maybe. But keep in mind old browser support, in worst case we need to gracefully degrade the functionality.

Is there a way to generate "global" CSS to override default size of the sidebar? And resizer may just "control" this CSS.

IMHO the sidebar shall not depend and control its parents; a resizer alone may notify (via dispatch event) that it's position changed and app will adjust its UI then.

**Review by yurydelendik — CHANGES_REQUESTED** [review]
move pdf_sidebar_resizer into its own file.

**Comment by Snuffleupagus** [comment]
> Are we OK with rewriting the sidebar CSS using flexbox?

Considering that we've not yet officially rescinded support for older IE versions, that may not work too well; see e.g. the following compatibility data https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Flexible_Box_Layout/Using_CSS_flexible_boxes#Browser_compatibility.

Also, looking again at the screen-shot in https://github.com/mozilla/pdf.js/pull/8957#issuecomment-332390324,  I do wonder if we'd need to try and reach consensus on and implement #4052 first. Otherwise I'm not sure how you'd ever be able to dynamically update all of the media-queries involved!?

**Comment by brendandahl** [comment]
> Are we OK with rewriting the sidebar CSS using flexbox?

It's probably about time we re-visit our supported browser/version. The can-i-use for flex box is around 97%.

FWIW: the original mock-up of the PDF viewer was done in flexbox and it was SO much cleaner than what we ended up with. Unfortunately, the site doesn't work any more http://people.mozilla.org/~shorlander/mockups/PDF.js/PDFjs-Tour-01/index.html

**Comment by brendandahl** [comment]
Shorlander moved the original mock http://design.firefox.com/people/shorlander/pdfjs/prototype/pdfjs-ui-demo-i01.html

**Comment by ironyman** [comment]
I noticed some weirdness in the code, the naming here is pretty confusing, should we fix this?

https://github.com/mozilla/pdf.js/blob/master/web/viewer.js#L60

**Comment by timvandermeij** [comment]
Closing since this is superseded by #9026. This solution is a bit more complicated than that one. Nevertheless, thank you for working on this as it provided us with ideas on how to improve the current situation and implement this feature more quickly!

**Pull request closed without merging**